### PR TITLE
[bitnami/jenkins] Fix: InitScripts, InitHookScripts

### PR DESCRIPTION
**Pull request by EC-Sol**
### Description of the change

Because not existing template about initScripts, initHookScripts 
So add init-scripts-configmap, init-hook-scripts-configmap

### Benefits

Now we can use values initScripts, initScriptCM, initScriptSecret, initHookScripts, initHookScriptCM, initHookScriptSecret

### Possible drawbacks

Nothing

### Applicable issues

- fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/jenkins/templates/deployment.yaml [hunk 1]
```diff
@@ -532,15 +536,15 @@ spec:
             sources:
               {{- if .Values.initScripts }}
               - configMap:
-                  name: {{ template "jenkins.initScripts" . }}
+                  name: {{ template "common.names.fullname" . }}-init-scripts
```
Comment (juan131) [inline bitnami/jenkins/templates/deployment.yaml:539]: Let's define the `jenkins.initScripts` in the [_helpers.tpl](https://github.com/bitnami/charts/blob/main/bitnami/jenkins/templates/_helpers.tpl) instead:
```
{{/*
Get the initialization scripts ConfigMap name.
*/}}
{{- define "jenkins.initdbScriptsCM" -}}
{{- if .Values.initScriptsCM -}}
  {{- printf "%s" (tpl .Values.initScriptsCM $) -}}
{{- else -}}
  {{- printf "%s-init-scripts" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
```
Then, you can simplify this like:
```yaml
              {{- if or .Values.initScripts .Values.initScriptsCM }}
              - configMap:
                  name: {{ template "jenkins.initdbScriptsCM" . }}
              {{- end }}
```

File: bitnami/jenkins/templates/deployment.yaml [hunk 2]
```diff
@@ -550,15 +554,15 @@ spec:
             sources:
               {{- if .Values.initHookScripts }}
               - configMap:
-                  name: {{ template "jenkins.initHookScripts" . }}
+                  name: {{ template "common.names.fullname" . }}-init-hook-scripts
```
Comment (juan131) [inline bitnami/jenkins/templates/deployment.yaml:557]: Same suggestions can be applied here.

File: bitnami/jenkins/templates/init-hook-scripts-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.initHookScripts }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "common.names.fullname" . }}-init-hook-scripts
```
Comment (juan131) [inline bitnami/jenkins/templates/init-hook-scripts-configmap.yaml:10]:
```suggestion
  name: {{ printf "%s-init-hook-scripts" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
```

File: bitnami/jenkins/templates/init-script-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.initScripts }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "common.names.fullname" . }}-init-scripts
```
Comment (juan131) [inline bitnami/jenkins/templates/init-script-configmap.yaml:10]:
```suggestion
  name: {{ printf "%s-init-scripts" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
```

File: bitnami/jenkins/templates/init-hook-scripts-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+data:
+  {{- range $filename, $content := .Values.initHookScripts }}
+  {{ $filename }}: |
+    {{- $content | nindent 4 }}
```
Comment (juan131) [inline bitnami/jenkins/templates/init-hook-scripts-configmap.yaml:19]: Should we support templates here and use `common.tplvalues.render` instead?

File: bitnami/jenkins/templates/init-hook-scripts-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  {{- end }}
+{{- end }}
```
Comment (juan131) [inline bitnami/jenkins/templates/init-hook-scripts-configmap.yaml:21]: Please fix the "no newline at end of file" warning

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this fix!! I wonder when we broke this feature in the past.

File: bitnami/jenkins/templates/deployment.yaml (hunk 1 above)
Comment (EC-Sol) [reply bitnami/jenkins/templates/deployment.yaml]: @juan131  I accept this review.

File: bitnami/jenkins/templates/deployment.yaml (hunk 2 above)
Comment (EC-Sol) [reply bitnami/jenkins/templates/deployment.yaml]: I applied this.

File: bitnami/jenkins/templates/init-hook-scripts-configmap.yaml (hunk 5 above)
Comment (EC-Sol) [reply bitnami/jenkins/templates/init-hook-scripts-configmap.yaml:19]: I'm not very good at English, and I've just barely moved beyond the beginner level with Helm charts, so I don't understand what you're talking about.

File: bitnami/jenkins/templates/_helpers.tpl [hunk 7]
```diff
@@ -126,3 +126,25 @@ Check if there are rolling tags in the images
...
+
+{{/*
+Get the initialization hook scripts ConfigMap name.
+*/}}
+{{- define "jenkins.initHookScriptsName" -}}
+{{- if .Values.initScriptsCM -}}
+  {{- printf "%s" (tpl .Values.initHookScriptsCM $) -}}
+{{- else -}}
+  {{- printf "%s-init-scripts" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
```
Comment (juan131) [inline bitnami/jenkins/templates/_helpers.tpl:149]: Typos:
```suggestion
{{- if .Values.initHookScriptsCM -}}
  {{- printf "%s" (tpl .Values.initHookScriptsCM $) -}}
{{- else -}}
  {{- printf "%s-init-hook-scripts" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" -}}
{{- end -}}
```

File: bitnami/jenkins/templates/deployment.yaml [hunk 8]
```diff
@@ -530,35 +534,27 @@ spec:
...
-              {{- if .Values.initHookScriptsCM }}
+              {{- if or .Values.initHookScripts .Values.initHookScriptsCM }}
               - configMap:
-                name: {{ template ".Values.initHookScriptsCM" . }}
+                  name: {{ template "jenkins.initHookScriptsName" . }}
               {{- end }}
               {{- if .Values.initHookScriptsSecret }}
               - secret:
-                  secretName: {{ template ".Values.initHookScriptsSecret" . }}
+                  secretName: {{ .Values.initHookScriptsSecret }}
```
Comment (juan131) [inline bitnami/jenkins/templates/deployment.yaml:557]: Admit templates:
```suggestion
                  secretName: {{ print (tpl .Values.initHookScriptsSecret .) }}
```

File: bitnami/jenkins/templates/init-script-configmap.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ printf "%s-init-scripts" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+data:
+  {{- range $filename, $content := .Values.initHookScripts }}
```
Comment (juan131) [inline bitnami/jenkins/templates/init-script-configmap.yaml:17]: Typo:
```suggestion
  {{- range $filename, $content := .Values.initScripts }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/jenkins/templates/deployment.yaml (hunk 8 above)
Comment (EC-Sol) [reply bitnami/jenkins/templates/deployment.yaml]: @juan131 
I have applied the suggested Admit template to both initScriptsSecret and initHookScriptsSecret. Please check the new commit.

**Pull request merged**